cmd/kubectl-gadget/utils/post-process.go

```diff
@@ -29,20 +29,23 @@ type PostProcess struct {
 
 type postProcessSingle struct {
 	orig             io.Writer
-	transform        func(string) string
+	Node             string
+	callback         func(line string, node string)
+	transform        func(line string) string
 	firstLinePrinted *uint64
 	buffer           string // buffer to save incomplete strings
 	skipFirstLine    bool
 	verbose          bool
 }
 
 type PostProcessConfig struct {
-	Flows         int                 // number of flow this should process
-	Transform     func(string) string // function to be called to transform the output before printing
-	OutStream     io.Writer           // stream to print the standard output
-	ErrStream     io.Writer           // stream to print the error output
-	SkipFirstLine bool                // only print the first line once
-	Verbose       bool                // Verbose mode
+	Flows         int                            // number of flow this should process
+	Callback      func(line string, node string) // function to be called each time there is new data from the pod
+	Transform     func(line string) string       // function to be called to transform the output before printing
```

Comment:
should document somewhere that `Transform()` is only used when `Callback` is nil, or call it too when Callback is defined

Reply:
Yes, added a comment to the Transform documentation.